### PR TITLE
Feature/ionic updated

### PR DESCRIPTION
**Pull request by cmreddy071**
-Created ionic search-bar example
- written test cases

### CODE REVIEW

**Comment by cmreddy071** [comment]
- Used the collapsible-appbar example, Fix the alignment issues.

File: package.json [hunk 1]
```diff
@@ -49,6 +52,12 @@
     "@types/jasminewd2": "~2.0.3",
     "@types/node": "~8.9.4",
     "codelyzer": "^5.0.0",
+    "cordova-plugin-device": "^2.0.2",
```
Comment (joebochill) [inline package.json]: plugins should be in dependencies not devdependencies

File: package.json [hunk 2]
```diff
@@ -61,5 +70,19 @@
...
+  "description": "An Ionic project",
+  "cordova": {
+    "plugins": {
+      "cordova-plugin-whitelist": {},
+      "cordova-plugin-statusbar": {},
+      "cordova-plugin-device": {},
+      "cordova-plugin-splashscreen": {},
+      "cordova-plugin-ionic-webview": {},
+      "cordova-plugin-ionic-keyboard": {}
+    },
```
Comment (joebochill) [inline package.json]: where is the config line that we added for data-list and collapsible-appbar?

File: src/global.scss [hunk 3]
```diff
@@ -24,3 +24,27 @@
...
+  height: 100%;
+  width: 100%;
+}
+
+ion-header,ion-toolbar {
+  color: map-get($pxb-white, 50);
+  --color : map-get($pxb-white, 50);
+  --background: map-get($pxb-blue, 500); 
+  background: map-get($pxb-blue, 500);
+  font-family: Open Sans, sans-serif;
```
Comment (joebochill) [inline src/global.scss]: Why do we have this if we are using the theme?

File: src/index.html [hunk 4]
```diff
@@ -7,19 +7,23 @@
...
   <meta name="msapplication-tap-highlight" content="no" />
 
   <link rel="icon" type="image/png" href="assets/icon/favicon.png" />
 
+  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+
   <!-- add to homescreen for ios -->
   <meta name="apple-mobile-web-app-capable" content="yes" />
   <meta name="apple-mobile-web-app-status-bar-style" content="black" />
+  <link href="https://fonts.googleapis.com/css?family=Roboto:300,400,500&display=swap" rel="stylesheet">
```
Comment (joebochill) [inline src/index.html]: Why are we importing roboto font?

**Comment by jeffvg** [comment]
Set focus on search field once method is called

**Comment by jeffvg** [comment]
update package to include "ANDROID_SUPPORT_ANNOTATIONS_VERSION": "27.+"

File: package.json (hunk 1 above)
Comment (cmreddy071) [reply package.json]: Moved to dependencies.

File: package.json (hunk 2 above)
Comment (cmreddy071) [reply package.json]: Added config line.

File: src/global.scss (hunk 3 above)
Comment (cmreddy071) [reply src/global.scss]: we don't have theme for ionic

File: src/index.html (hunk 4 above)
Comment (cmreddy071) [reply src/index.html]: Removed

File: src/global.scss (hunk 3 above)
Comment (joebochill) [reply src/global.scss]: we use the angular theme, which includes Open Sans

File: src/app/app.component.html [hunk 5]
```diff
@@ -1,3 +1,48 @@
 <ion-app>
-  <ion-router-outlet></ion-router-outlet>
-</ion-app>
+  <ion-header [class.hidden]="toggleSearch">
+    <ion-toolbar>
+      <div color="primary" class="main-header" fxLayout="row" fxLayoutAlign="space-between">
+        <button mat-icon-button>
+          <mat-icon class="menu-icon">menu</mat-icon>
+        </button>
```
Comment (joebochill) [inline src/app/app.component.html]: this should be an ion-button, refer to collapsible-appbar

File: src/app/app.component.html [hunk 6]
```diff
@@ -1,3 +1,48 @@
...
+      </div>
+    </ion-toolbar>
+  </ion-header>
+  <ion-header class="main-header search-block" [class.active]="toggleSearch">
+    <ion-toolbar class="search-toolbar">
+      <div color="primary" class="main-header" fxLayout="row" fxLayoutAlign="space-between">
+        <button class="search-icon" mat-icon-button disabled>
+          <mat-icon>search</mat-icon>
+        </button>
```
Comment (joebochill) [inline src/app/app.component.html]: should be ion-button

File: src/app/app.component.html [hunk 7]
```diff
@@ -1,3 +1,48 @@
...
+        <button mat-icon-button>
+          <mat-icon class="menu-icon">menu</mat-icon>
+        </button>
+        <div class="title-wrapper">
+          <div class="mat-title">President</div>
+          <div class="sub-title">Leader of the Free World</div>
+        </div>
+        <button mat-icon-button flex="end" (click)="openSearch()">
+          <mat-icon>search</mat-icon>
+        </button>
```
Comment (joebochill) [inline src/app/app.component.html]: should be ion-button

File: src/app/app.component.html [hunk 8]
```diff
@@ -1,3 +1,48 @@
...
+      <div color="primary" class="main-header" fxLayout="row" fxLayoutAlign="space-between">
+        <button class="search-icon" mat-icon-button disabled>
+          <mat-icon>search</mat-icon>
+        </button>
+        <div class="title-wrapper">
+          <input class="search-control" type="text" placeholder="Search" [(ngModel)]="searchText" #searchbar>
+        </div>
+        <button mat-button mat-icon-button (click)="searchClose()" style="color:rgba(0, 0, 0, 0.26)">
+          <mat-icon>close</mat-icon>
+        </button>
```
Comment (joebochill) [inline src/app/app.component.html]: should be ion-button

File: src/app/app.component.scss [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+.menu-icon {
+  margin-left: 10px;
+  height: 24px;
+}
+.search-icon {
+  margin-left: 10px;
+  margin-left: 0px;
+}
```
Comment (joebochill) [inline src/app/app.component.scss]: the positioning on these is strange...should be 16px on the left and 32px on the right. Use the ion-button and follow the implementation that was used on previously approved examples (e.g., collapsible-appbar)

File: src/app/app.component.scss [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+.title-wrapper {
+  flex: 1;
+  padding-left:32px; 
+  z-index: 10;
+  .mat-title,.sub-title {
+    font-size: 16px;
+    text-align: left;
+    line-height: 100%;
+    margin-bottom:2px;
```
Comment (joebochill) [inline src/app/app.component.scss]: why do we have this bottom margin?

File: src/app/app.component.scss [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+  }
+  .search-block {
+    .title-wrapper {
+      padding-left:22px; 
+    }
+    .search-icon {
+      padding-left:14px;
+    }
+  }
+}
```
Comment (joebochill) [inline src/app/app.component.scss]: we don't do conditional styling based on the OS

File: src/app/app.component.scss [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    &::-ms-clear {
+      height: 0;
+      width: 0;
+    }
+  }
+}
+.data-list {
+  margin-top: calc(50px);
+  margin-top: calc(35px + constant(safe-area-inset-top));
+  margin-top: calc(35px + env(safe-area-inset-top));
```
Comment (joebochill) [inline src/app/app.component.scss]: the header should be 56px...why is the margin for the list 50 or 35?

File: src/app/app.component.ts [hunk 13]
```diff
@@ -22,6 +31,15 @@ export class AppComponent {
...
     });
   }
+  openSearch() {
+    this.toggleSearch = true;
+    this.searchbar.nativeElement.focus();
+  }
+  searchClose() {
+    this.toggleSearch = false;
+    this.searchText = '';
+  }
```
Comment (joebochill) [inline src/app/app.component.ts]: you can combine these into a single function and say if(this.togglesearch).....focus()

**Comment by jeffvg** [comment]
not seeing issue with alignments as mentioned above
ionic-updated-android

**Comment by cmreddy071** [comment]
Updated themes version to 2.2.4

File: src/app/app.component.html (hunk 6 above)
Comment (cmreddy071) [reply src/app/app.component.html]: changed to ion-button

File: src/app/app.component.html (hunk 5 above)
Comment (cmreddy071) [reply src/app/app.component.html]: changed to ion-button

File: src/app/app.component.html (hunk 7 above)
Comment (cmreddy071) [reply src/app/app.component.html]: changed to ion-button

File: src/app/app.component.html (hunk 8 above)
Comment (cmreddy071) [reply src/app/app.component.html]: changed to ion-button

File: src/app/app.component.scss (hunk 9 above)
Comment (cmreddy071) [reply src/app/app.component.scss]: Changed

File: src/app/app.component.scss (hunk 10 above)
Comment (cmreddy071) [reply src/app/app.component.scss]: removed bottom-margin

File: src/app/app.component.scss (hunk 11 above)
Comment (cmreddy071) [reply src/app/app.component.scss]: removed styles based on the OS.

File: src/app/app.component.scss (hunk 12 above)
Comment (cmreddy071) [reply src/app/app.component.scss]: Changed to 56px.

**Comment by cmreddy071** [comment]
Requested changes are done.

**Pull request merged**